### PR TITLE
Lordpidey devil fixes

### LINKED ISSUE #27403: A couple of devil issues
1. Switching from human to blood lizard will cause you to have duplicates of all your abilities.
2. Infernal jaunt allows you to press the appear button next to a possible contractee and then move away so you can appear anywhere

### LINKED ISSUE #28051: Devil Bugs
[Note]: #15794, #18520

True devils/Archdevils are completely deaf. This makes it extremely difficult to, you know, talk people into taking contracts.
True devils get two sets of every button, meaning two hellfires and two pitchforks at the same time.
True/Archdevils do not heal at all.
True/Archdevils are immune to burns from fire, but still take burn damage from lasers (?)
True/Archdevils can be handcuffed.
True/Archdevils can be cremated. This completely defeats the purpose of being a huge hulking thing if you can be stuffed in the cremator like a changeling.
Accompanying the previous point, if your body gets incinerated, you spawn shortly afterwards in Maintenance as a deaf True Devil, with tier 1 powers.

You can shove contracts, signed and unsigned, into papertrays where they'll never ever be found.
People walk hellfire off like it's nothing smh

### PR DESCRIPTION
**Pull request by lordpidey**
:cl: Lordpidey
fix: True and arch devils are no longer deaf.
fix: Fixed some edge cases with devils getting two sets of spells.
fix: True/arch devils can now instabreak cuffs.
add: The codex gigas now indicates if a devil is ascendable or not.
tweak: Hellfire has been buffed, it now explodes multiple times.
/:cl:

why: fixes a couple of bugs.  I'm so sorry about devils being deaf.

fixes #28051
fixes #27403

### CODE REVIEW

**Comment by DoctoreMad** [comment]
>add: The codex gigas now indicates if a devil is ascendable or not.

Wait what? Is this just a feature that wasn't reported by the Codex Gigas or did you make some random devils unascendable?

**Comment by lordpidey** [comment]
It was added in #24693.

Event devils are unascendable
Gamemode devils are ascendable
Adminfuckery devils can be either, there's a button for each.

**Comment by lordpidey** [comment]
Marking this as WIP, as I was prevented from testing this today.  Sorry.

File: code/modules/spells/spell_types/devil.dm [hunk 1]
```diff
@@ -132,6 +128,14 @@
 		return
 	revert_cast()
 
+/obj/effect/proc_holder/spell/targeted/infernal_jaunt/proc/valid_location(mob/living/user = usr)
+	if(istype(get_area(user), /area/shuttle/)) // Can always phase in in a shuttle.
+		return TRUE
+	else
+		for(var/mob/living/C in orange(2, get_turf(user.loc))) //Can also phase in when nearby a potential buyer.
```
Comment (AnturK) [inline code/modules/spells/spell_types/devil.dm]: Can just be get_turf(user) here.

File: code/modules/projectiles/projectile/magic.dm [hunk 2]
```diff
@@ -412,3 +412,9 @@
 	exp_light = -1
 	exp_flash = 4
 	exp_fire= 5
+
+/obj/item/projectile/magic/aoe/fireball/infernal/on_hit(target)
+	. = ..()
```
Comment (AnturK) [inline code/modules/projectiles/projectile/magic.dm]: Is the original explosion still happening here intended ?

Comment (lordpidey) [inline code/modules/projectiles/projectile/magic.dm]: Yes, it doubles the fire effect for the first half second.

**Comment by AnturK** [comment]
Is this ready now ?

**Comment by lordpidey** [comment]
No, sorry.  I'm on a network that I can't port forward on, so testing is difficult.

**Comment by Cyberboss** [comment]
@lordpidey

**Comment by lordpidey** [comment]
Yeah, I'm sorry.  I'll try to get my roommate to help me test it tomorrow.

**Comment by lordpidey** [comment]
Did a bit more testing, fixed duplicate spells being given out.
